internal/k8s/exec.go

```diff
@@ -170,14 +171,33 @@ func ReturnDeploymentObject(clientset *kubernetes.Clientset, matchLabel string,
 
 	log.Info().Msgf("waiting for %s Deployment to be created", matchLabelValue)
 
-	// Create watch operation
-	objWatch, err := clientset.
-		AppsV1().
-		Deployments(namespace).
-		Watch(context.Background(), deploymentListOptions)
-	if err != nil {
-		log.Error().Msgf("error when attempting to search for Deployment: %s", err)
-		return nil, err
+	var objWatch watch.Interface
+	var err error
+
+	timeout := 20 * time.Second
+	startTime := time.Now()
+
+	// Create watch operation with retries
+	for {
+		// Attempt to watch the deployment
+		objWatch, err = clientset.AppsV1().Deployments(namespace).Watch(context.Background(), deploymentListOptions)
```

Comment:
wdyt if you check before the err nil :
``` 
for {
    objWatch, err = clientset.AppsV1().Deployments(namespace).Watch(context.Background(), deploymentListOptions)

    if err == nil {
        break
    }

    if time.Since(startTime) > timeout {
        log.Printf("Timeout reached while watching for Deployment")
        return nil, fmt.Errorf("timeout reached while watching for Deployment")
    }

    log.Printf("Error when attempting to watch Deployment: %s. Retrying...", err)
    time.Sleep(1 * time.Second)
}
```

Reply:
Yes this is correct ig cause it will pass for the last check as well